homeassistant/components/vegehub/sensor.py

```diff
@@ -0,0 +1,210 @@
+        )
+
+        hass.data[DOMAIN][sensor.unique_id] = sensor
+
+        sensors.append(sensor)
+
+    if sensors:
+        async_add_entities(sensors)
+
+
+class VegeHubSensor(SensorEntity):
+    """Class for VegeHub Analog Sensors."""
+
+    def __init__(
+        self,
+        name: str,
+        mac_address: str,
+        slot: int,
+        ip_addr: str,
+        dev_name: str,
+        data_type: str,
+        chan_type: str,
+    ) -> None:
+        """Initialize the sensor."""
+        new_id = (
+            f"vegehub_{mac_address}_{slot}".lower()
+        )  # Generate a unique_id using mac and slot
+
+        self._attr_name: str = name
+        self._data_type: str = data_type
+        self._unit_of_measurement: str = ""
+        self._attr_native_value = None
+
+        if chan_type == CHAN_TYPE_BATTERY:
+            self._unit_of_measurement = UnitOfElectricPotential.VOLT
+            self._attr_device_class = SensorDeviceClass.VOLTAGE
+        elif data_type == OPTION_DATA_TYPE_CHOICES[1]:
+            self._unit_of_measurement = PERCENTAGE
+            self._attr_device_class = SensorDeviceClass.MOISTURE
+        elif data_type == OPTION_DATA_TYPE_CHOICES[2]:
+            self._unit_of_measurement = UnitOfTemperature.CELSIUS
+            self._attr_device_class = SensorDeviceClass.TEMPERATURE
+        else:
+            self._unit_of_measurement = UnitOfElectricPotential.VOLT
+            self._attr_device_class = SensorDeviceClass.VOLTAGE
+
+        self._mac_address: str = mac_address
+        self._slot: int = slot
+        self._attr_unique_id: str = new_id
+        self.entity_id: str = "sensor." + new_id
+        self._ip_addr: str = ip_addr
+        self._dev_name: str = dev_name
+
+    @property
+    def name(self) -> str:
+        """Return the name of the sensor."""
+        return self._attr_name
+
+    @property
+    def native_value(self) -> float | None:
+        """Return the state of the sensor."""
+        if (
+            self._data_type == OPTION_DATA_TYPE_CHOICES[1] and self._attr_native_value
+        ):  # Percentage
+            return VH400_transform(self._attr_native_value)
+        if (
+            self._data_type == OPTION_DATA_TYPE_CHOICES[2] and self._attr_native_value
+        ):  # Temperature C
+            if isinstance(self._attr_native_value, (int, str, float)):
+                return (41.6700 * float(self._attr_native_value)) - 40.0000
+            return None
+
+        if isinstance(self._attr_native_value, (int, str, float)):
+            return float(self._attr_native_value)
+        return None
+
+    @property
+    def suggested_unit_of_measurement(self) -> str:
+        """Return the unit of measurement of this entity."""
+        return self._unit_of_measurement
+
+    @property
+    def native_unit_of_measurement(self) -> str:
+        """Return the unit of measurement of this entity."""
+        return self._unit_of_measurement
+
+    @property
```

Comment:
remove any where you're just returning the _attr....

Reply:
Fixed in commit 166f2c9db70cd478e4aaafbf52c366b802ca4781.